presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -0,0 +1,227 @@
+    {
+        return queryPlan.getPlanNodeCosts().entrySet().stream()
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private Map<PlanNodeId, PlanNodeCost> getExecutionCostsById(QueryStats queryStats)
+    {
+        //for some reason, there are duplicate PlanNodeId-s in operatorSummaries.
+        BinaryOperator<PlanNodeCost> mergeOnlyIfEqual = (a, b) -> {
+            assertEquals(a, b);
+            return b;
+        };
+        return queryStats.getOperatorSummaries().stream()
+                .collect(toMap(OperatorStats::getPlanNodeId, this::toPlanNodeCost, mergeOnlyIfEqual));
+    }
+
+    private PlanNodeCost toPlanNodeCost(OperatorStats operatorStats)
+    {
+        return PlanNodeCost.builder()
+                .setOutputRowCount(new Estimate(operatorStats.getOutputPositions()))
+                .setOutputSizeInBytes(new Estimate(operatorStats.getOutputDataSize().toBytes()))
+                .build();
+    }
+
+    private Optional<Double> asOptional(Estimate estimate)
+    {
+        return estimate.isValueUnknown() ? Optional.empty() : Optional.of(estimate.getValue());
+    }
+
+    private void outputSummary(Map<MetricComparison.Result, List<MetricComparison>> resultSummaries, MetricComparison.Result result)
+    {
+        System.out.println(format("[%s]\t-\t[%s]", result, resultSummaries.getOrDefault(result, emptyList()).size()));
+    }
+
+    static class MetricComparison
+    {
+        final PlanNode planNode;
+        final String metricName;
+        final Optional<Double> estimatedCost;
+        final Optional<Double> executionCost;
+
+        public MetricComparison(PlanNode planNode, String metricName, Optional<Double> estimatedCost, Optional<Double> executionCost)
+        {
+            this.planNode = planNode;
+            this.metricName = metricName;
+            this.estimatedCost = estimatedCost;
+            this.executionCost = executionCost;
+        }
+
+        @Override
+        public String toString()
+        {
+            return format("Metric [%s] - [%s] - estimated: [%s], real: [%s] - plan node: [%s]",
+                    metricName, result(), print(estimatedCost), print(executionCost), planNode);
+        }
+
+        Result result()
+        {
+            return estimatedCost
+                    .map(estimate -> executionCost
+                            .map(execution -> estimateMatchesReality(estimate, execution) ? MATCH : DIFFER)
+                            .orElse(NO_BASELINE))
+                    .orElse(NO_ESTIMATE);
+        }
+
+        private String print(Optional<Double> cost)
+        {
+            return cost.map(Object::toString).orElse("UNKNOWN");
+        }
+
+        private boolean estimateMatchesReality(Double estimate, Double execution)
+        {
+            return Math.abs(execution - estimate) / execution < 0.1;
+        }
+
+        enum Result
+        {
+            NO_ESTIMATE,
```

Comment:
because those correspond to the cases, when there was 'no estimate' and 'no baseline'?